src/mpi/coll/include/coll_types.h

```diff
@@ -0,0 +1,114 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2006 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ *
+ *  Portions of this code were written by Intel Corporation.
+ *  Copyright (C) 2011-2017 Intel Corporation.  Intel provides this material
+ *  to Argonne National Laboratory subject to Software Grant and Corporate
+ *  Contributor License Agreement dated February 8, 2012.
+ */
+#ifndef MPIR_COLL_TYPES_H_INCLUDED
+#define MPIR_COLL_TYPES_H_INCLUDED
+
+#include "queue.h"
+#include "uthash.h"
+
+#define MPIR_COLL_FLAG_REDUCE_L 1
+#define MPIR_COLL_FLAG_REDUCE_R 0
+
+enum {
+    TREE_TYPE_KNOMIAL=0,
+    TREE_TYPE_KARY
+};
+
+/* enumerator for all the collective operations */
+enum {
+    BCAST = 0,
+    REDUCE,
+    ALLREDUCE,
+    SCATTER,
+    GATHER,
+    ALLGATHER,
+    ALLTOALL,
+    ALLTOALLV,
+    ALLTOALLW,
+    BARRIER,
+    REDUCESCATTER,
+    SCAN,
+    EXSCAN,
+    GATHERV,
+    ALLGATHERV,
+    SCATTERV
```

Comment:
It might be nice to alphabetize these so it's easier to make sure we don't miss one and/or need to add one later.

Reply:
Done